File: app/models/ElectronicAddressArg.java
/*
 * ElectronicAddressArg.java
 */


package models;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Id;
import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.ManyToOne;
import javax.persistence.MapsId;
import javax.persistence.JoinColumn;
import javax.persistence.Basic;

import play.db.ebean.Model;


/**
 * ElectronicAddressArg
 */
@Entity
@Table(name = "tbl_electronic_address_args")
public class ElectronicAddressArg extends Model {

    @Id
    @Column(name = "electronic_address_arg_id")
    @GeneratedValue
    public Integer electronic_address_arg_id;

    @ManyToOne
    @MapsId
    @JoinColumn(name = "electronic_address_id")
    ElectronicAddress _electronicAddress;

    @Basic
    @Column(name = "name", length = 512)
    public String name;

    @Basic
    @Column(name = "required")
    public boolean required;

    @Basic
    @Column(name = "type", length = 512)
    public String type;

    @Basic
    @Column(name = "use_", length = 512)
    public String use;
}

File: app/models/dto/ElectronicAddressArg.scala
package models.dto

import play.api.libs.json.Json

/** ElectronicAddressArg DTO (Data Transfer Object)
  *
  * Entity class storing rows of table ElectronicAddressArgs.
  *
  * @param electronicAddressArgId Database column ELECTRONIC_ADDRESS_ARG_ID PrimaryKey
  * @param electronicAddressId Database column ELECTRONIC_ADDRESS_ID
  * @param name Database column NAME
  * @param required Database column REQUIRED
  * @param `type` Database column TYPE
  * @param use Database column USE
  */
case class ElectronicAddressArg(
    electronicAddressArgId: Option[Int],
    electronicAddressId: Int,
    name: Option[String],
    required: Option[Boolean],
    `type`: Option[String],
    use: Option[String])

object ElectronicAddressArg {
  implicit val electronicAddressArgReads = Json.reads[ElectronicAddressArg]
  implicit val electronicAddressArgWrites = Json.writes[ElectronicAddressArg]
}
